src/main/java/io/github/opencubicchunks/cubicchunks/chunk/CubeMap.java

```diff
@@ -1,24 +1,125 @@
 package io.github.opencubicchunks.cubicchunks.chunk;
 
-import java.util.Set;
-import java.util.concurrent.ConcurrentHashMap;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.IntFunction;
+import java.util.function.IntSupplier;
 
-public class CubeMap {
-    private final ConcurrentHashMap.KeySetView<Integer, Boolean> loadedCubes = ConcurrentHashMap.newKeySet();
+import javax.annotation.Nullable;
 
-    public boolean isLoaded(int cubeY) {
-        return loadedCubes.contains(cubeY);
+import com.mojang.datafixers.util.Either;
+import io.github.opencubicchunks.cubicchunks.chunk.cube.CubeStatus;
+import io.github.opencubicchunks.cubicchunks.chunk.cube.LevelCube;
+import io.github.opencubicchunks.cubicchunks.chunk.util.CubePos;
+import io.github.opencubicchunks.cubicchunks.utils.Coords;
+import it.unimi.dsi.fastutil.longs.LongSet;
+import net.minecraft.core.SectionPos;
+import net.minecraft.server.level.ChunkHolder;
+import net.minecraft.server.level.ServerChunkCache;
+import net.minecraft.server.level.ServerPlayer;
+import net.minecraft.world.level.chunk.ChunkStatus;
+
+public interface CubeMap {
```

Comment:
I sort of like the `I` prefix, though in this case as we're matching vanilla maybe it's fine?

Reply:
I would like to move away from the I prefix as it is non standard in java and its not used in mc code on forge and fabric anymore